src/modm/driver/pwm/tlc594x_impl.hpp

```diff
@@ -15,40 +15,39 @@
 #endif
 
 // ----------------------------------------------------------------------------
-template<uint16_t CHANNELS, typename Spi, typename Xlat, typename Vprog, typename Xerr>
-uint8_t modm::TLC594X<CHANNELS, Spi, Xlat, Vprog, Xerr>::status[CHANNELS*3/2];
-template<uint16_t CHANNELS, typename Spi, typename Xlat, typename Vprog, typename Xerr>
-uint8_t modm::TLC594X<CHANNELS, Spi, Xlat, Vprog, Xerr>::gs[CHANNELS*3/2];
-template<uint16_t CHANNELS, typename Spi, typename Xlat, typename Vprog, typename Xerr>
-uint8_t modm::TLC594X<CHANNELS, Spi, Xlat, Vprog, Xerr>::dc[CHANNELS*3/4];
-
-// ----------------------------------------------------------------------------
-template<uint16_t CHANNELS, typename Spi, typename Xlat, typename Vprog, typename Xerr>
-void
-modm::TLC594X<CHANNELS, Spi, Xlat, Vprog, Xerr>::initialize(uint16_t channels, uint8_t dots, bool writeCH, bool writeDC)
+template<uint16_t CHANNELS, typename Spi, typename Xlat, typename Xblank, typename Vprog, typename Xerr>
+modm::TLC594X<CHANNELS, Spi, Xlat, Xblank, Vprog, Xerr>::TLC594X(uint16_t channels, uint8_t dots, bool writeCH, bool writeDC, bool enable)
```

Comment:
`initialize()` was intentionally not the constructor. You'll have issues with this when statically allocating the driver, since the constructor will be called *before* main, and thus *before* you have connected/initialized the SPI peripheral. The GPIO will work (since we [explicitly enable them](https://modm.io/reference/module/modm-platform-gpio/#enable_ports) even before the static constructors), but the SPI calls may not do anything, or worse, even hang forever.

Reply:
Moved initialization from constructor to separate method.